airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionUpdaterWorkflowImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling;
+
+import io.airbyte.config.StandardSyncInput;
+import io.airbyte.scheduler.models.IntegrationLauncherConfig;
+import io.airbyte.scheduler.models.JobRunConfig;
+import io.airbyte.workers.temporal.sync.SyncWorkflow;
+import io.temporal.api.enums.v1.ParentClosePolicy;
+import io.temporal.workflow.CancellationScope;
+import io.temporal.workflow.ChildWorkflowOptions;
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ConnectionUpdaterWorkflowImpl implements ConnectionUpdaterWorkflow {
+
+  private boolean canStart;
+  private boolean isRunning = false;
+  private boolean isDeleted = false;
+  private final boolean skipScheduling = false;
+  // private Optional<SyncWorkflow> childSync = Optional.empty();
+  CancellationScope syncWorkflowCancellationScope = CancellationScope.current();
+
+  public ConnectionUpdaterWorkflowImpl() {
+    canStart = false;
+  }
+
+  public ConnectionUpdaterWorkflowImpl(final ConnectionUpdaterWorkflowConfig connectionUpdaterWorkflowConfig) {
+    canStart = !connectionUpdaterWorkflowConfig.isFirstStart();
+  }
+
+  @Override
+  public SyncResult run() {
+    Workflow.await(() -> canStart());
+
+    // TODO: bmoric get time to wait through an activity
+    final Duration timeToWait = Duration.ofSeconds(5);
+
+    Workflow.await(timeToWait, () -> skipScheduling());
+
+    // TODO: Fetch config (maybe store it in GCS)
+    log.info("Starting child WF");
+
+    final SyncWorkflow childSync = Workflow.newChildWorkflowStub(SyncWorkflow.class,
+        ChildWorkflowOptions.newBuilder()
+
+            // This will cancel the child workflow when the parent is terminated
```

Comment:
Is this true? According to https://community.temporal.io/t/parent-and-child-workflow-close-policy/463 it looks like `PARENT_CLOSE_POLICY_ABANDON` would allow the child workflow to keep running instead of cancelling. 

I think what we actually want is `PARENT_CLOSE_POLICY_REQUEST_CANCEL`?

Reply:
Thanks for the catch, I have moved it to `PARENT_CLOSE_POLICY_TERMINATE` since we don't really care about gracefully finishing all the updates state update will be done in the parent wf.